internal/exec/engine.go

```diff
@@ -274,8 +276,21 @@ func (e *Engine) fetchReferencedConfig(cfgRef types.ConfigReference) (types.Conf
 	if err != nil {
 		return types.Config{}, err
 	}
+	headers := http.Header{}
+	for headerName, headerValue := range resource.ConfigHeaders {
+		headers[headerName] = headerValue
+	}
+	if cfgRef.HTTPHeaders != nil && len(cfgRef.HTTPHeaders) > 0 {
+		for _, headerData := range cfgRef.HTTPHeaders {
+			if len(headerData) != 2 {
+				e.Logger.Crit("Malformed HTTP header: %v", headerData)
+				return types.Config{}, fmt.Errorf("malformed HTTP header: %v", headerData)
+			}
+			headers.Add(strings.TrimSpace(string(headerData[0])), strings.TrimSpace(string(headerData[1])))
```

Comment:
I'm not thrilled about the redundancy between this and `HTTPHeaders.Parse()`.  Could both be moved into a `HTTPHeaders` helper method that modifies an existing `http.Header`?

Reply:
I decided to keep the logic here, but started using HTTPHeaders.Parse() to get rid of the code duplication